MapboxNavigation/NavigationViewController.swift

```diff
@@ -561,6 +561,14 @@ extension NavigationViewController: RouteControllerDelegate {
         }
         return advancesToNextLeg
     }
+    
+    @objc public func routeController(_ routeController: RouteController, didEnterTunnelAt location: CLLocation?) {
+        if let _ = location {
+            self.styleManager.applyStyle(NightStyle())
```

Comment:
Do you think we could first check if the style needs to be updated before continually updating the style on every location update?

Reply:
@bsudekum I do a safety check before I apply the style in `applyStyle(_:)` on `line #110`.

Reply:
Also, this update is only applied to tunnel locations within a tunnel intersection.